src/Runner.Worker/JobRunner.cs

```diff
@@ -206,6 +217,59 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
             }
         }
 
+        private async Task<TaskResult> CompleteJobAsync(IRunServer runServer, IJobServer jobServer, IExecutionContext jobContext, Pipelines.AgentJobRequestMessage message, TaskResult? taskResult = null)
```

Comment:
Should we just pass in a single IRunnerService for the server we intend to return here, and provide overloads for it?

Its kind of weird to pass in two variables when one is expected to be null

Reply:
they both don't share common props  :( `IJobServer` and `IRunServer` are different)...
Any alternatives to make it clean? I did have two methods for their specific calls, but I need a some way of abstracting that, so I created that common method that takes both, else I have to have `if else` every where we call this, which is bit disturbing to see...

Reply:
Like we call `await CompleteJobAsync` 5 times and they are common to both flows, so 🤷‍♂️

Reply:
Definitely a super nit suggestion so feel free to ignore

But they both inherit from IJobServer iirc.

Base case could throw, overloads of IJobServer and IRunServer do as expected, and when you call it c# should automatically call the more detailed overload if one exists. Or you could just pass in a single IRunService and check the type and fail if its not an expected type.

Reply:
addressed!